solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -73,7 +78,19 @@ contract InterchainAccountRouter is Router, IInterchainAccountRouter {
         external
         returns (bytes32)
     {
-        return _dispatch(_destinationDomain, abi.encode(msg.sender, calls));
+        return _dispatch(_destinationDomain, calls.format(msg.sender));
+    }
+
+    /**
+     * @notice Dispatches a sequence of calls to be relayed by the sender's interchain account on the destination domain.
+     * @param _destinationDomain The domain of the chain where the message will be sent to.
+     * @param calls The sequence of calls to be relayed.
+     */
+    function dispatch(
```

Comment:
Four flavors of dispatch feels like a lot, thoughts on just having two, one with CallWithValue[] and the other with CallWithValue?

Reply:
the raw interaction from tools like the explorer motivated this

Reply:
also trying to avoid backwards incompatible changes